src/components/TextInput.tsx

```diff
@@ -4,33 +4,58 @@ import {
   createRestyleComponent,
   VariantProps,
   createVariant,
-  createBox,
 } from '@shopify/restyle'
-import { TextInput as RNTextInput } from 'react-native'
+import { TextInput } from 'react-native'
 import tinycolor from 'tinycolor2'
+import { SvgProps } from 'react-native-svg'
 import { Color, theme, Theme } from '../theme/theme'
 import { useColors, useInputVariants } from '../theme/themeHooks'
+import Box from './Box'
+import Text from './Text'
+import TouchableOpacityBox from './TouchableOpacityBox'
 
-const TextInputBox = createBox<Theme, React.ComponentProps<typeof RNTextInput>>(
-  RNTextInput,
-)
-
-const TextInput = createRestyleComponent<
-  VariantProps<Theme, 'inputVariants'> &
-    React.ComponentProps<typeof TextInputBox>,
+const BoxWrapper = createRestyleComponent<
+  VariantProps<Theme, 'inputVariants'> & React.ComponentProps<typeof Box>,
   Theme
->([createVariant({ themeKey: 'inputVariants' })], TextInputBox)
+>([createVariant({ themeKey: 'inputVariants' })], Box)
 
-type Props = React.ComponentProps<typeof TextInput> & {
+type Props = React.ComponentProps<typeof BoxWrapper> & {
   placeholderTextColor?: Color
   textColor?: Color
   fontSize?: number
+  fontWeight?:
```

Comment:
In my experience font weight doesn't work consistently across platforms. 
See this issue for more info <https://github.com/facebook/react-native/issues/26193>

I've been using different fonts for the weights.

In the wireless app I added a bunch of text variants so that we don't need to mess around with weights...
```ts
  textVariants: {
    ...textVariants,

    buttonLight: { ...textVariants.button, ...textVariants.light },
    buttonMedium: { ...textVariants.button, ...textVariants.medium },
    buttonBold: { ...textVariants.button, ...textVariants.bold },

    body0Light: { ...textVariants.body0, ...textVariants.light },
    body0Medium: { ...textVariants.body0, ...textVariants.medium },
    body0Bold: { ...textVariants.body0, ...textVariants.bold },

    body1Light: { ...textVariants.body1, ...textVariants.light },
    body1Medium: { ...textVariants.body1, ...textVariants.medium },
    body1Bold: { ...textVariants.body1, ...textVariants.bold },

    body2Light: { ...textVariants.body2, ...textVariants.light },
    body2Medium: { ...textVariants.body2, ...textVariants.medium },
    body2Bold: { ...textVariants.body2, ...textVariants.bold },

    body3Light: { ...textVariants.body3, ...textVariants.light },
    body3Medium: { ...textVariants.body3, ...textVariants.medium },
    body3Bold: { ...textVariants.body3, ...textVariants.bold },

    h1Light: { ...textVariants.h1, ...textVariants.light },
    h1Medium: { ...textVariants.h1, ...textVariants.medium },
    h1Bold: { ...textVariants.h1, ...textVariants.bold },

    h2Light: { ...textVariants.h2, ...textVariants.light },
    h2Medium: { ...textVariants.h2, ...textVariants.medium },
    h2Bold: { ...textVariants.h2, ...textVariants.bold },

    h3Light: { ...textVariants.h3, ...textVariants.light },
    h3Medium: { ...textVariants.h3, ...textVariants.medium },
    h3Bold: { ...textVariants.h3, ...textVariants.bold },

    h4Light: { ...textVariants.h4, ...textVariants.light },
    h4Medium: { ...textVariants.h4, ...textVariants.medium },
    h4Bold: { ...textVariants.h4, ...textVariants.bold },
  },
  
  const textVariants = {
  body0: {
    color: 'primaryText',
    fontFamily: Font.regular,
    fontSize: 21,
  },
  body1: {
    color: 'primaryText',
    fontFamily: Font.regular,
    fontSize: 17,
  },
  body1Light: {
    color: 'primaryText',
    fontFamily: Font.light,
    fontSize: 17,
  },
  body2: {
    color: 'primaryText',
    fontFamily: Font.regular,
    fontSize: 14,
  },
  body3: {
    color: 'primaryText',
    fontFamily: Font.regular,
    fontSize: 12,
  },
  bold: {
    fontFamily: Font.bold,
  },
  h0: {
    color: 'primaryText',
    fontFamily: Font.regular,
    fontSize: 90,
  },
  h1: {
    color: 'primaryText',
    fontFamily: Font.regular,
    fontSize: 54,
  },
  h2: {
    color: 'primaryText',
    fontFamily: Font.regular,
    fontSize: 48,
  },
  h3: {
    color: 'primaryText',
    fontFamily: Font.regular,
    fontSize: 44,
  },
  h4: {
    color: 'primaryText',
    fontFamily: Font.regular,
    fontSize: 36,
  },
  h5: {
    color: 'primaryText',
    fontFamily: Font.regular,
    fontSize: 28,
  },
  light: {
    fontFamily: Font.light,
  },
  medium: {
    fontFamily: Font.medium,
  },
  regular: {
    fontFamily: Font.regular,
  },
  subtitle1: {
    color: 'primaryText',
    fontFamily: Font.regular,
    fontSize: 24,
  },
  subtitle2: {
    color: 'primaryText',
    fontFamily: Font.regular,
    fontSize: 20,
  },
  subtitle3: {
    color: 'primaryText',
    fontFamily: Font.regular,
    fontSize: 18,
  },
  subtitle4: {
    color: 'primaryText',
    fontFamily: Font.regular,
    fontSize: 14,
  },
  button: {
    color: 'primaryText',
    fontFamily: Font.medium,
    fontSize: 20,
  },
}
  ```

Reply:
Removed